Joystream/joystream
# Dev init vstore

Updated the `dev-init` script to initialize versioned store using call information produced in https://github.com/Joystream/versioned-store-js/pull/3/files#diff-05499f43b3181eeacb4730a497f134c7R194

Updated storage-node README

You should now be able to interact with the media app fully as in production network.

To test:
make sure you build the joystream-node docker image with `scripts/build-joystream-node-docker-image.sh`
then run the following script:

```sh
storage-node/scripts/run-dev-instance.sh
```

you should see something like:
```sh
Removing network devchain-and-ipfs-node_default
WARNING: Network devchain-and-ipfs-node_default not found.
Removing volume devchain-and-ipfs-node_ipfs-data
WARNING: Volume devchain-and-ipfs-node_ipfs-data not found.
Removing volume devchain-and-ipfs-node_chain-data
WARNING: Volume devchain-and-ipfs-node_chain-data not found.
Creating network "devchain-and-ipfs-node_default" with the default driver
Creating volume "devchain-and-ipfs-node_ipfs-data" with local driver
Creating volume "devchain-and-ipfs-node_chain-data" with local driver
Creating devchain-and-ipfs-node_chain_1 ... done
Creating devchain-and-ipfs-node_ipfs_1  ... done
yarn run v1.22.4
$ /Users/mokhtar/joystream/joystream/node_modules/.bin/storage-cli dev-init
Secp256k1 bindings are not compiled. Pure JS implementation will be used.
  joystream:storage-cli:dev Ensuring Alice is sudo +0ms
Running setup
  joystream:storage-cli:dev Ensuring Alice is as member.. +48ms
  joystream:storage-cli:dev Registering Alice as member.. +50ms
  joystream:storage-cli:dev Setting Alice as content working group lead +727ms
  joystream:storage-cli:dev Initializing Content Directory +6s
  joystream:storage-cli:dev creating classes... +1ms
  joystream:storage-cli:dev processing 13 transactions... +0ms
  joystream:storage-cli:dev 1/13 processed. +0ms
  joystream:storage-cli:dev 2/13 processed. +3ms
  joystream:storage-cli:dev 3/13 processed. +1ms
  joystream:storage-cli:dev 4/13 processed. +1ms
  joystream:storage-cli:dev 5/13 processed. +1ms
  joystream:storage-cli:dev 6/13 processed. +1ms
  joystream:storage-cli:dev 7/13 processed. +0ms
  joystream:storage-cli:dev 8/13 processed. +1ms
  joystream:storage-cli:dev 9/13 processed. +1ms
  joystream:storage-cli:dev 10/13 processed. +1ms
  joystream:storage-cli:dev 11/13 processed. +0ms
  joystream:storage-cli:dev 12/13 processed. +1ms
  joystream:storage-cli:dev 13/13 processed. +0ms
  joystream:storage-cli:dev adding schemas to classes... +6s
  joystream:storage-cli:dev processing 13 transactions... +0ms
  joystream:storage-cli:dev 1/13 processed. +0ms
  joystream:storage-cli:dev 2/13 processed. +2ms
  joystream:storage-cli:dev 3/13 processed. +1ms
  joystream:storage-cli:dev 4/13 processed. +1ms
  joystream:storage-cli:dev 5/13 processed. +0ms
  joystream:storage-cli:dev 6/13 processed. +1ms
  joystream:storage-cli:dev 7/13 processed. +1ms
  joystream:storage-cli:dev 8/13 processed. +1ms
  joystream:storage-cli:dev 9/13 processed. +1ms
  joystream:storage-cli:dev 10/13 processed. +1ms
  joystream:storage-cli:dev 11/13 processed. +0ms
  joystream:storage-cli:dev 12/13 processed. +1ms
  joystream:storage-cli:dev 13/13 processed. +1ms
  joystream:storage-cli:dev creating entities... +6s
  joystream:storage-cli:dev processing 5 transactions... +1ms
  joystream:storage-cli:dev 1/5 processed. +0ms
  joystream:storage-cli:dev 2/5 processed. +47ms
  joystream:storage-cli:dev 3/5 processed. +2ms
  joystream:storage-cli:dev 4/5 processed. +2ms
  joystream:storage-cli:dev 5/5 processed. +24ms
  joystream:storage-cli:dev Setting Local development node as bootstrap endpoint +13s
  joystream:storage-cli:dev Transferring tokens to storage role account +4s
  joystream:storage-cli:dev Making Alice the storage Lead +3ms
({"member_id":"u64","role_account_id":"AccountId","reward_relationship":"Option<u64>","role_stake_profile":"Option<{\"stake_id\":\"u64\",\"termination_unstaking_period\":\"Option<u32>\",\"exit_unstaking_period\":\"Option<u32>\"}>"},Linkage<u64>):: Input doesn't match output, received 0x000000007da26da5e784569ae3cd4c8558852c82d69fa904bd1a14611cd3fd15c79335d40000000000000000, created 0x0000000000000000d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d0000
  joystream:storage-cli:dev Making //Colossus account a storage provider +18s
  joystream:storage-cli:dev created new storage opening: 1 +6s
  joystream:storage-cli:dev applied with application id: 1 +6s
  joystream:storage-cli:dev filling storage opening +1ms
({"member_id":"u64","role_account_id":"AccountId","reward_relationship":"Option<u64>","role_stake_profile":"Option<{\"stake_id\":\"u64\",\"termination_unstaking_period\":\"Option<u32>\",\"exit_unstaking_period\":\"Option<u32>\"}>"},Linkage<u64>):: Input doesn't match output, received 0x000000007da26da5e784569ae3cd4c8558852c82d69fa904bd1a14611cd3fd15c79335d40000000000000000, created 0x0000000000000000d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d0000
  joystream:storage-cli:dev Assigned storage provider id: 1 +6s

  Chain is setup with Dev storage provider:
    providerId = 1
    roleAccountId = 5FNPVC9sBT4ns61t6WTCvcsHm438YkZ2o96NixNZ2P1J4vhj
    roleKey = //Colossus
  
✨  Done in 74.40s.
yarn run v1.22.4
$ /Users/mokhtar/joystream/joystream/node_modules/.bin/colossus --dev
Secp256k1 bindings are not compiled. Pure JS implementation will be used.
  joystream:runtime:base Init +0ms
  joystream:runtime:identities Init +0ms
  joystream:runtime:identities Error loading account file: The "path" argument must be of type string. Received undefined +2ms
  joystream:runtime:balances Init +0ms
  joystream:runtime:roles Init +0ms
  joystream:runtime:assets Init +0ms
  joystream:runtime:discovery Init +0ms

  Chain is setup with Dev storage provider:
    providerId = 1
    roleAccountId = 5FNPVC9sBT4ns61t6WTCvcsHm438YkZ2o96NixNZ2P1J4vhj
    roleKey = //Colossus
  
Secp256k1 bindings are not compiled. Pure JS implementation will be used.
________                     _____                               
______(_)__________  __________  /__________________ _______ ___ 
_____  /_  __ \_  / / /_  ___/  __/_  ___/  _ \  __ `/_  __ `__ \
____  / / /_/ /  /_/ /_(__  )/ /_ _  /   /  __/ /_/ /_  / / / / /
___  /  \____/_\__, / /____/ \__/ /_/    \___/\__,_/ /_/ /_/ /_/ 
/___/         /____/                                             
  joystream:sync Starting sync run... +0ms
  joystream:colossus announcing public url +0ms
Starting API server...
API server started. { address: '::', family: 'IPv6', port: 3001 }
  joystream:sync sync run complete +392ms
  joystream:storage:storage IPFS node is up with identity: QmTQGheuHQafvwrrubJ63eApoGVckoEcXX71DE7TmiLsxo +0ms
  joystream:discovery:publish adding service info file to node +0ms
  joystream:discovery:publish publishing... +51ms
  joystream:discovery:publish {
  joystream:discovery:publish   name: 'QmTQGheuHQafvwrrubJ63eApoGVckoEcXX71DE7TmiLsxo',
  joystream:discovery:publish   value: '/ipfs/QmcSjtVMfDSSNYCxNAb9PxNpEigCw7h1UZ77gip3ghfbnA'
  joystream:discovery:publish } +2m
  joystream:runtime:base Got nonce for 5FNPVC9sBT4ns61t6WTCvcsHm438YkZ2o96NixNZ2P1J4vhj from chain: 0 +2m
  joystream:runtime:base Signing and sending tx +0ms
  joystream:runtime:base TX status: Ready +9ms
  joystream:runtime:base TX Ready. +0ms
  joystream:runtime:base TX status: Finalized +2s
  joystream:runtime:base TX Finalized. +0ms
  joystream:colossus publishing complete, scheduling next update +2m
```

wait a few minutes to see successful publishing of ipns identity,
then point your pioneer to the local node:

```
yarn workspace pioneer start
```

Create a channel, and upload a video...
